lib/Parser.js

```diff
@@ -1059,16 +1059,18 @@ class Parser extends Tapable {
 	walkCallExpression(expression) {
 		let result;
 
-		function walkIIFE(functionExpression, options) {
-			const params = functionExpression.params;
-			const args = options.map(function(arg) {
-				const renameIdentifier = this.getRenameIdentifier(arg);
-				if(renameIdentifier && this.applyPluginsBailResult1("can-rename " + renameIdentifier, arg)) {
-					if(!this.applyPluginsBailResult1("rename " + renameIdentifier, arg))
+		function walkIIFE(functionExpression, options, currentThis) {
+			function renameArgOrThis(argOrThis) {
+				const renameIdentifier = this.getRenameIdentifier(argOrThis);
+				if(renameIdentifier && this.applyPluginsBailResult1("can-rename " + renameIdentifier, argOrThis)) {
+					if(!this.applyPluginsBailResult1("rename " + renameIdentifier, argOrThis))
 						return renameIdentifier;
 				}
-				this.walkExpression(arg);
-			}, this);
+				this.walkExpression(argOrThis);
+			}
+			const params = functionExpression.params;
+			const args = options.map(renameArgOrThis, this);
+			const renameThis = currentThis ? renameArgOrThis.call(this, currentThis) : null;
```

Comment:
Could you switch this line with the previous line? So that arguments are evaluated in the correct order.